docs/design/mp_overview.md

```diff
@@ -0,0 +1,186 @@
+mandatory that an Operator updates the data for its Object when it is
+in the Provisioned state and optional in any other state. Operators
+may also mutate the state of objects other than its Stateful Object,
+but never store these objects data. We call these Objects Mutable
+objects from an Operator perspective. The Objects types with respects
+to Operators will become clear as we detail Mizar's workflow.
+
+The specification of an Operator's Stateful Object, Mutable Object,
+and Mutating Actions are what defines the management plane workflow.
+
+### Inter-Operators Communications
+
+Operators communicate with each other to evolve the object's state.
+When an operator mutates the state of the object it invokes the next
+operator(s) in the object pipeline. The invocation is done through a
+callback mechanism. According to the deployment and specific
+implementation the callback is translated into one of multiple modes.
+In all of these modes, the operator developer only focuses on the
+workflow through calling the callback function without worrying about
+the details of the communication implementation.
+
+1. **Through the API Server(s)**: In this mode, operators communication is
+   implicit with the help of the API server. As one operator makes the
+   callback to invoke the next step of the object pipeline, the
+   only changes the object state and update the object rhough the API
+   server. As the next Operator in the pipeline handles the object in
+   the new state, communication happens implicitly. The drawback of
+   this approach is the loading on the API server, the latency
+   involved to evolve the object's pipeline. The advantage is that the
+   object state is always tracked through the system.
+1. **Direct Service Communication**: In this mode, operators exposes a
+   service interface through RPC or any other mechanism, and the
+   callback effectively calls the RPC on the other operator. The
+   calling operators must handle failure scenarios in such case. The
+   drawback of this approach is the complexity of handling failure
+   scenarios. Also, as the intermediate states of the objects are not
+   updates to the API server, it becomes harder to troubleshoot
+   problems in the object-pipeline. This approach __may__ have latency
+   advantage, especially if the API server communication is a
+   bottlneck. However, if the API server and kubernetes storage scales
+   horizontally latency of the this approach and the second approach
+   may be comparable. In that case, this approac only adds
+   implementation complexitie that are hard to diagnose.
+1. **Simple Function Calls**: In this mode, all operators are deployed
+   into a single (horizontally scalable) operator. The functional
+   implementation of the operators is still seperated through
+   call-back interfaces. But the callbacks are implemented as simply
+   function calls. This approach is the simplist to develop, test, and
```

Comment:
This approach is the **simplest** to develop

Reply:
ok